lib/config/genconfig/awshaprovisionconfig.go

```diff
@@ -0,0 +1,1287 @@
+	if err != nil {
+		fmt.Println("ERRORs in Validation:\n", err)
+		return nil
+	}
+
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptAwsConfig() (err error) {
+	err = c.PromptProfile()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptRegion()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptVpcId()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptCidr()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptSshKeyPairName()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptAmi()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptDeleteOnTermination()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptLoadBalancerAccessLogs()
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptCidr() (err error) {
+	isCidrBlockAddr, err := c.HasCidrBlockAddr()
+	if err != nil {
+		return
+	}
+
+	if !isCidrBlockAddr {
+		err1 := c.PromptPrivateSubnet()
+		if err1 != nil {
+			return err1
+		}
+		err1 = c.PromptPublicSubnet()
+		if err1 != nil {
+			return err1
+		}
+	}
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptProfile() (err error) {
+	profile, err := c.Prompt.InputStringDefault("AWS Profile", "default")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().Profile = profile
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptRegion() (err error) {
+	awsRegions := AwsRegionsImpFactory(c.Prompt)
+	awsRegion, err := awsRegions.Choose("AWS Region")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().Region = awsRegion
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptVpcId() (err error) {
+	vpcId, err := c.Prompt.InputStringRequired("AWS VPC ID")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().AwsVpcID = vpcId
+	return
+}
+
+func (c *AwsHaProvisionConfig) HasCidrBlockAddr() (isCiderBlock bool, err error) {
+	isCiderBlock, err = c.Prompt.Confirm("Do you want to use AWS CIDR Block", "yes", "no")
+	if err != nil {
+		return
+	}
+	if isCiderBlock {
+		err = c.PromptCidrBlockAddr()
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptCidrBlockAddr() (err error) {
+	ciderBlockAddr, err := c.Prompt.InputStringRegex("AWS CIDR Block Address", IP_REGEX)
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().AwsCidrBlockAddr = ciderBlockAddr
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptPrivateSubnet() (err error) {
+	privateSubnet1, err := c.Prompt.InputStringRequired("AWS Private Subnet 1:")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().PrivateCustomSubnets = []string{privateSubnet1}
+
+	privateSubnet2, err := c.Prompt.InputStringRequired("AWS Private Subnet 2:")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().PrivateCustomSubnets = append(c.Config.InitAws().InitConfigAwsSettings().PrivateCustomSubnets, privateSubnet2)
+
+	privateSubnet3, err := c.Prompt.InputStringRequired("AWS Private Subnet 3:")
+	if err != nil {
+		return
+	}
```

Comment:
We can add checks if all public or private subnets in array are unique

Reply:
I will consider this as feature enhancement for next time.